File: metadata/repository/v1/files/ruby/3/poly.rb
require 'json';

def json_to_request(input)
  json = JSON.parse(input);

  Request.new(
    json['url'],
    json['body'],
    json['method'],
    json['headers'],
    json['params']
  )
end

class Request
  attr_reader :url, :body, :method, :headers, :params

  def initialize(url, body, method, headers, params)
    @url = url
    @body = body
    @method = method
    @headers = headers
    @params = params
  end
end

class Response
  attr_accessor :body, :status_code, :headers

  def initialize(body, status_code = 200, headers = {})
    @body = body
    @status_code = status_code
    @headers = headers
  end

  def self.ok(body)
    new(body)
  end

  def self.created(body)
    new(body, 201)
  end

  def self.not_found(body)
    new(body, 404)
  end

  def print_headers
    serialized = "{"

    headers.each do |k, v|
      serialized = "#{serialized} \"#{k}\": \"#{v}\""
    end

    "#{serialized} }"
  end

  def to_s
    JSON.generate({
      data: body,
      status: status_code,
      headers: headers,
      kv: {}
    })
  end
end

File: metadata/repository/v1/files/python/3/poly.py
import sys, json;

class Cache:
    store = {}

    @classmethod
    def init(cls, kv):
        cls.store = kv

    @classmethod
    def dump(cls):
        return cls.store

    @classmethod
    def get(cls, key):
        return cls.store.get(key)

    @classmethod
    def set(cls, key, value):
        cls.store[key] = str(value)

class Request:
    def __init__(self, input):
        self.method = input["method"]
        self.url = input["url"]
        self.body = input["body"]
        self.headers = input["headers"]
        self.params = input["params"]

        # Init the cache
        Cache.init(input["kv"])

class Response:
    def __init__(self, body):
        self.body = body
        self.status_code = 200
        self.headers = {}

    def to_json(self):
        res = {
            'data': self.body,
            'status': self.status_code,
            'headers': self.headers,
            'kv': Cache.dump()
        }
        return json.dumps(res)
